pkg/resources/custom_diffs_test.go

```diff
@@ -1082,3 +1081,40 @@ func Test_RecreateWhenSecretTypeChangedExternallyForOAuth2(t *testing.T) {
 		})
 	}
 }
+
+func Test_RecreateWhenSecondaryConnectionChangedExternally(t *testing.T) {
+	tests := []struct {
+		name              string
+		expectedIsPrimary string
+		stateValue        map[string]string
+	}{
+		{
+			name:              "changed from is_primary from false to true",
+			expectedIsPrimary: "false",
+			stateValue: map[string]string{
+				"is_primary": "true",
+			},
```

Comment:
Add more tests cases:
- going from true to false
- no changes (it would be probably caught in acc tests, but just in case) true -> true and false -> false

Reply:
this will be further tested in next PR, as well as `ParseCommaSeparatedAccountIdentifierArray()`